docs/migration_guide_v2.md

```diff
@@ -0,0 +1,632 @@
+* [`scw version`](#scw-version)
+* [`scw wait`](#scw-wait)
+
+## Justification for a v2
+
+### Design Docker oriented
+
+CLI v1 was designed to offer a syntax close to the Docker syntax.
+For instance, running a command such as `echo foobar` on a remote server is `scw run ubuntu-bionic echo foobar` which mimic `docker run ubuntu echo foobar`.
+While this can be useful for some tasks, there are plenty of actions that don't fit in this paradigm.
+For instance, attaching to a running server sub-resources such as volumes or security groups are not performed easily using this paradigm because Docker doesn't provide the same features.
+
+In CLI v2 we offer support for a wide range of actions on all resources present and coming in the Scaleway Elements ecosystem.
+Actions are organized around a set of verbs such as `list`, `get`, 'create', 'update' that can be used with a wide variety of products and do not suppose any preconceived workflow.
+
+### No design for multiple products
+
+CLI v1 was created at a time targeting a single Scaleway Elements product: instance.
+Scaleway got many more products in its Elements ecosystem that need to be available from the CLI.
+Having ad-hoc commands for all our products was not going to be a scalable solution.
+We needed to have a more systematic approach across all products.
+
+In CLI v2, we have a common syntax across all our products: service sub-resource verb.
+As an example:
+
+```shell
+# v2
+scw instance server list
+```
+
+* **instance**: Refers to a resource API
+* **server**: Refers to a sub-resource maintained by this API
+* **list**: Refers to a verb apply to the currently selected API
+
+In English, it would be: "list all servers available on the instance API".
+
+### No automated code generation
+
+CLI v1 didn't have any code generation features to easily create supports in all SDKs and developer tools we support.
+We invested in our code generation features to be able to synchronize support and fixes across all our tools:
+
+* [scaleway-cli](https://github.com/scaleway/scaleway-cli)
+* [scaleway-sdk-go](https://github.com/scaleway/scaleway-sdk-go)
+* [scaleway provider for Terraform](https://github.com/terraform-providers/terraform-provider-scaleway/)
+
+### Old unmaintained dependencies
+
+CLI v1 required a lot of dependencies that are not actively maintained anymore.
+As a team, we don't have the bandwidth to maintain all the dependencies, perform regular audits and open-source management required to make all those dependencies thrive.
+With the CLIv2, we want to minimize them as much as possible and focus only on well-supported external libraries when required.
+
+### Few tests
+
+CLI v1 didn't have high test coverage and no test generation that could be inferred from an underlying SDK.
+CLI v2 builds on top of the tests infrastructure of the code generation we have to increase test coverage.
+We also support different types of tests: unit test, conformance test, End to End testing.
+
+## Commands
```

Comment:
The title should explain that what is listed below are `commands equivalences`, not just `commands`